src/detect-engine-event.c

```diff
@@ -56,6 +58,70 @@ static void DetectEngineEventFree (DetectEngineCtx *, void *);
 void EngineEventRegisterTests(void);
 #endif
 
+static bool PrefilterEventIsPrefilterable(const Signature *s, int smtype)
+{
+    const SigMatch *sm;
+    for (sm = s->init_data->smlists[DETECT_SM_LIST_MATCH]; sm != NULL; sm = sm->next) {
+        switch (sm->type) {
+            case DETECT_ENGINE_EVENT:
+                return true;
+        }
+    }
+    return false;
+}
+static bool PrefilterEngineEventIsPrefilterable(const Signature *s)
+{
+    return PrefilterEventIsPrefilterable(s, DETECT_ENGINE_EVENT);
+}
+
+static bool PrefilterDecodeEventIsPrefilterable(const Signature *s)
+{
+    return PrefilterEventIsPrefilterable(s, DETECT_DECODE_EVENT);
+}
+
+static void PrefilterPacketEventSet(PrefilterPacketHeaderValue *v, void *smctx)
```

Comment:
unrelated to your PR but perhaps a nice addition: There's no explanation what the diff fields in the union `PrefilterPacketHeaderValue;` are. I'm only guessing looking at other code..

Reply:
My understanding is that they are free-forms integers that keywords can use for setting up their prefilter...